execution/registry/registry.go

```diff
@@ -16,38 +16,112 @@ package registry
 
 import (
 	"fmt"
+	"net"
 
 	"github.com/hyperledger/burrow/crypto"
 )
 
-func AuthorizedPeersProvider(state IterableReader) func() ([]string, []string) {
-	return func() ([]string, []string) {
-		var peerIDs, peerAddrs []string
+type NodeStats struct {
+	Addresses map[string]map[crypto.Address]struct{}
+}
+
+func NewNodeStats() NodeStats {
```

Comment:
On the one hand I like that it rhymes with `AccountStats` and has the same lifecycle on the other hand it's not really stats... probably fine.